platform/linux-generic/pool/subsystem.c

```diff
@@ -15,3 +19,219 @@ ODP_SUBSYSTEM_CONSTRUCTOR(pool)
 	odp_subsystem_constructor(pool);
 }
 
+int odp_pool_init_global(void)
+{
+	pool_module_t *mod;
+
+	mod = odp_subsystem_active_module(pool, mod);
+	if (mod == NULL) {
+		ODP_ERR("No active module in pool subsystem\n");
+		return -1;
+	}
+	if (mod->base.init_global == NULL) {
+		ODP_ERR("No defined init_global function "
+			"in module %s of pool subsystem\n", mod->base.name);
+		return -1;
+	}
```

Comment:
Yes, I leave it to user programmer, so different policies can be impled, like:

1) IF a subsystem assumes all its api pointers should not be NULL, it can bypass some checks.
2) IF a subsystem assumes at least one active module exist, it can bypass some checks.
3) a subsystem can write these checks in subsystem->init_global() to assure active module exists and/or api pointers are not NULL.
4) in least case, user programmer can write inline/macro to check per their subsystem's requirements.